viewer/viewer.js

```diff
@@ -5179,111 +5437,159 @@ function buildConnections(req, res, cb) {
 
     connects[linkId].value++;
     updateValues(f, connects[linkId], fields);
-  }
+  } // process
 
-  buildSessionQuery(req, function(bsqErr, query, indices) {
-    if (bsqErr) {
-      return cb(bsqErr, 0, 0, 0);
-    }
-    query.query.bool.filter.push({exists: {field: req.query.srcField}});
-    query.query.bool.filter.push({exists: {field: req.query.dstField}});
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  // processResultSets - process the hits of each search resultset into nodesHash and connects
+  function processResultSets (connResultSets, cb) {
```

Comment:
refactored some of the code processing each query result `graph` into this local function. it processes an array of 1..2 result sets recursively in the same way the other functions I mentioned above do.